### PR TITLE
Specify maximum amount of memory available (Nitro)

### PR DESCRIPTION
**Pull request by gbryant-arm**
Add maximum isolate memory to policy (Nitro only for now).
Addresses https://github.com/veracruz-project/veracruz/issues/280

### CODE REVIEW

File: test-collateral/generate-policy/src/main.rs [hunk 1]
```diff
@@ -169,6 +172,7 @@ impl Arguments {
             enclave_debug_mode: false,
             execution_strategy: String::new(),
             enable_clock: false,
+            max_memory: 0,
```
Comment (dreemkiller) [inline test-collateral/generate-policy/src/main.rs:175]: Would we rather set a rational default value here?
Also, (this is a question up for discussion, not a demand): Do we want to put the units in the field name:
```
max_memory_mb
```
or something or other?

File: policy-utils/src/policy.rs [hunk 2]
```diff
@@ -120,6 +122,7 @@ impl Policy {
         execution_strategy: ExecutionStrategy,
         file_hashes: Vec<FileHash>,
         enable_clock: bool,
+        max_memory: u32,
```
Comment (dreemkiller) [inline policy-utils/src/policy.rs:125]: Maybe instead of `max_memory` make it `max_memory_mb` so users/policy readers/writers aren't left wondering

Comment (gbryant-arm) [reply policy-utils/src/policy.rs:125]: Good idea

File: test-collateral/generate-policy/src/main.rs (hunk 1 above)
Comment (gbryant-arm) [reply test-collateral/generate-policy/src/main.rs:175]: > Would we rather set a rational default value here?

Hmm like a fraction?

Comment (dreemkiller) [reply test-collateral/generate-policy/src/main.rs:175]: ha ha!

Like "256" - which is what we currently use.

**Review by dreemkiller — DISMISSED** [review]

File: test-collateral/generate-policy/src/main.rs (hunk 1 above)
Comment (gbryant-arm) [reply test-collateral/generate-policy/src/main.rs]: Haha got it :+1:

**Review by dreemkiller — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
Seems to have been a timeout of one of the IceCap tests.  I've rerun the failing CI build target.

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
+1+1 = +1 reached, merging.

**Pull request merged**